### PR TITLE
Feature: file search

### LINKED ISSUE #9: Feature: file search

### PR DESCRIPTION
**Pull request by fallenoak**
First stab at searching. Open to suggestions on naming. Would `search`, `searchFiles`, or `findFiles` be better than `find`?

Closes #9

### CODE REVIEW

File: src/spec/mpq.spec.js [hunk 1]
```diff
@@ -94,4 +94,78 @@ describe('MPQ', () => {
...
+
+      mpq.close();
+    });
+
+    test('throws if calling find on closed mpq', async () => {
+      const mpq = await MPQ.open('/fixture/vanilla-standard.mpq');
+
+      mpq.close();
+
+      expect(() => { mpq.find('*'); }).toThrow(Error);
```
Comment (timkurvers) [inline src/spec/mpq.spec.js]: Might be better for readability to put the code under test on a new line (like an indented closure), but nothing major.

File: src/lib/mpq.mjs [hunk 2]
```diff
@@ -18,6 +18,39 @@ class MPQ {
...
+
+    if (findHandle.isNull()) {
+      const errno = StormLib.GetLastError();
+
+      if (errno === StormLib.ERROR_NO_MORE_FILES) {
+        return [];
+      } else {
+        throw new Error(`Find failed (error ${errno})`);
+      }
+    }
```
Comment (timkurvers) [inline src/lib/mpq.mjs:39]: Ah neat! I think this was actually missing from Blizzardry, or?

File: src/spec/mpq.spec.js [hunk 3]
```diff
@@ -94,4 +94,78 @@ describe('MPQ', () => {
...
+      expect(result).toEqual({
+        'fileName': 'fixture.txt',
+        'plainName': 'fixture.txt',
+        'hashIndex': 3886,
+        'blockIndex': 0,
+        'fileSize': 13,
+        'compSize': 21,
+        'fileTimeLo': 414638976,
+        'fileTimeHi': 30643794,
+        'locale': 0
```
Comment (timkurvers) [inline src/spec/mpq.spec.js]: Any specific reason for these properties being quoted?

**Review by timkurvers — APPROVED** [review]
Neat! Async/await making the tests beautiful to read.

**Comment by timkurvers** [comment]
Hm, given `MPQ.prototype.openFile`, you might expect seeing `File(s)` in the method name for searching. I think this was one of the main reasons why Blizzardry currently has a `files` object, so that it doesn't get awkward with method names. I personally think `search` covers it pretty nicely., though.

While we're at it, can we move the `MPQ.open` method to a static method inside of the class? Or is there a reason it's living outside currently?

**Comment by fallenoak** [comment]
@timkurvers Renamed `find` to `search`!

Regarding `MPQ.open`: I think I missed the boat... did static functions show up in ES2015? I know static properties still are in proposal form, but after Googling a bit, it seems that maybe static functions are already a thing?

Edit: Apparently I knew they were a thing when I added a few to `wowser-math`. I need to remember to eat more fish. I'll tackle updating that in a separate commit on master.

File: src/spec/mpq.spec.js (hunk 3 above)
Comment (fallenoak) [reply src/spec/mpq.spec.js]: Copy pasta! I brought the shape over from `EmStormLib.cpp`.

Update: fixed!

File: src/lib/mpq.mjs (hunk 2 above)
Comment (fallenoak) [reply src/lib/mpq.mjs:39]: Yeah! I think Blizzardry returns an empty array in all cases where the find functions fail. One of the things I'm loving about `embind` is how easy it is to make constants from the headers (eg. `ERROR_NO_MORE_FILES`) available in JS. I just mark them in the bindings and voila, they're available on `StormLib`.

File: src/spec/mpq.spec.js (hunk 1 above)
Comment (fallenoak) [reply src/spec/mpq.spec.js]: I'll switch this to the non-curly-brace version: `() => mpq.find('*')`.

Update: switched!

File: src/lib/mpq.mjs (hunk 2 above)
Comment (timkurvers) [reply src/lib/mpq.mjs:39]: Ah that's super great indeed 👍

**Comment by timkurvers** [comment]
No worries, excellent stuff 👍

**Pull request merged**